docs/src/content/docs/fr/reference/plugins.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Référence des modules d'extension
+description: Une vue d'ensemble de l'API des modules d'extension Starlight.
+tableOfContents:
+  maxHeadingLevel: 4
+---
+
+Les modules d'extension Starlight peuvent personnaliser la configuration, l'interface utilisateur et le comportement de Starlight, tout en étant faciles à partager et à réutiliser.
+Cette page de référence documente l'API à laquelle ces modules d'extension ont accès.
+
+Consultez la [référence de configuration](/fr/reference/configuration/#plugins) pour en savoir plus sur l'utilisation d'un module d'extension Starlight.
+
+## Référence rapide de l'API
+
+Un module d'extension Starlight a la forme suivante.
+Voir ci-dessous pour plus de détails sur les différentes propriétés et paramètres des hooks.
+
+```ts
+interface StarlightPlugin {
+  name: string;
+  hooks: {
+    setup: (options: {
+      config: StarlightUserConfig;
+      updateConfig: (newConfig: StarlightUserConfig) => void;
+      addIntegration: (integration: AstroIntegration) => void;
+      astroConfig: AstroConfig;
+      command: 'dev' | 'build' | 'preview';
+      isRestart: boolean;
+      logger: AstroIntegrationLogger;
+    }) => void | Promise<void>;
+  };
+}
+```
+
+## `name`
+
+**Type :** `string`
+
+Un module d'extension doit fournir un nom unique qui le décrit. Le nom est utilisé lors de [l'affichage des messages](#logger) liés à ce module d'extension et peut être utilisé par d'autres modules d'extension pour détecter la présence de ce dernier.
+
+## `hooks`
+
+Les hooks sont des fonctions que Starlight appelle pour exécuter le code du module d'extension à des moments spécifiques. Actuellement, Starlight prend en charge un seul hook `setup`.
+
+### `hooks.setup`
+
+La fonction de configuration du module d'extension appelée lorsque Starlight est initialisé (pendant le hook [`astro:config:setup`](https://docs.astro.build/fr/reference/integrations-reference/#astroconfigsetup) de l'intégration).
+Le hook `setup` peut être utilisé pour mettre à jour la configuration de Starlight ou ajouter des intégrations Astro.
+
+Ce hook est appelé avec les options suivantes :
+
+#### `config`
+
+**Type :** `StarlightUserConfig`
+
+Une copie en lecture seule de la [configuration de Starlight](/fr/reference/configuration) fournie par l'utilisateur.
+Cette configuration peut avoir été mise à jour par d'autres modules d'extension configurés avant celui en cours.
+
+#### `updateConfig`
+
+**Type :** `(newConfig: StarlightUserConfig) => void`
+
+Une fonction de rappel pour mettre à jour la [configuration de Starlight](/fr/reference/configuration) fournie par l'utilisateur.
```

Comment:
[Reference](https://developer.mozilla.org/fr/docs/Glossary/Callback_function) for "fonction de rappel"